### DISCUSSION TITLE
What unit tests does Euphony use?

### DISCUSSION THREAD

**Discussion by happy-refactoring** [euphony-io/euphony#145] [Q&A] [answered]
Euphony is mainly using c++ language & java.
There are so many tools to use unit-test each language.

What unit tests does Euphony use?

I want to know the unit-test name and the link of example code!
could you let me know?

**Comment by DahyeonWoo** [comment, marked as answer]
In Euphony, for unit test, we use gtest for cpp and JUnit for java.

[C++]
Answer : gtest
Example : https://github.com/euphony-io/euphony/blob/54966925eb8a78d72e47eb31acd03c270f4c9694/euphony/src/main/cpp/tests/waveBuilderTest.cpp

[Java]
Answer: JUnit
Example: https://github.com/euphony-io/euphony-listener/blob/master/app/src/test/java/co/jbear/euphony_listener/ExampleUnitTest.java

**Reply by happy-refactoring** [reply]
Wow, thanks for your answer.
And.. what about java?

**Reply by happy-refactoring** [reply]
Thanks for your answer!! :)

**Comment by Kimhaedam** [comment]
Answer: JUnit
Example: https://github.com/euphony-io/euphony-listener/blob/master/app/src/test/java/co/jbear/euphony_listener/ExampleUnitTest.java

**Comment by hyunjung-choi** [comment]
Euphony uses gtest and JUnit4 for the unit test. 
![image](https://user-images.githubusercontent.com/69616347/178779419-4008f9a0-a28f-4857-88bf-b38f80d56dc7.png)
This folder is the unit test folder in Java.
You will be able to see the unit test in this folder.

The link is here.
https://github.com/euphony-io/euphony/tree/master/euphony/src/test/java/euphony/lib

**Reply by happy-refactoring** [reply]
Wow, I'm also wondering about JUnit's version! Thanks for giving me useful information!
